docs/reference/hooks.md

```diff
@@ -1166,7 +1166,7 @@ Called at the beginning of the create snippet view. Works in a similar way to `b
 
 ### `after_delete_snippet`
 
-Called when a Snippet is deleted. The callable passed into the hook will receive the model instance(s) as a queryset along with the request object. If the callable returns an `HttpResponse`, that response will be returned immediately to the user, and Wagtail will not proceed to call `redirect()` to the listing view.
+Called when a Snippet is deleted. The callable passed into the hook will receive the model instance(s) as a list along with the request object. If the callable returns an `HttpResponse`, that response will be returned immediately to the user, and Wagtail will not proceed to call `redirect()` to the listing view.
```

Comment:
This has never been consistent since the hooks were introduced: if there's only one snippet to be deleted, the hook is called with a list, and a queryset otherwise.

The base bulk action view doesn't store a reference to the queryset, it's immediately evaluated and the items are checked for permissions, so we can't call the hook with the queryset:

https://github.com/wagtail/wagtail/blob/0187a1494bc56bab4592dc0d3d314121edf590d0/wagtail/admin/views/bulk_action/base_bulk_action.py#L93-L107

I was thinking of storing the result in `self.queryset` before returning it in `get_queryset()`, but weirdly the method is a `classmethod`:

https://github.com/wagtail/wagtail/blob/0187a1494bc56bab4592dc0d3d314121edf590d0/wagtail/admin/views/bulk_action/base_bulk_action.py#L53-L55

I'm cautious to make any changes to the base bulk action view as it seems out of scope for this PR. Considering that this hook has been partially lost since 4.0, I think having this consistently be a list isn't so bad.

I considered adding a `.. versionchanged` here but I'm not sure what's the right way to document this, seeing that the hook has never been consistent, and the fact that it hasn't been accessible since 4.0.